R/qc_cvs.R

```diff
@@ -0,0 +1,147 @@
+#' @importFrom tidyr drop_na
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+#' @importFrom stats sd
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' qc_cvs(
+#' data,
+#' grouping = pep_stripped_sequence,
+#' condition = condition,
+#' intensity = fg_quantity,
+#' plot = TRUE,
+#' plot_style = "density"
+#' )
+#' }
+qc_cvs <-
+  function(data, grouping, condition, intensity, plot = TRUE, plot_style = "density")
+  {
+    if(plot == FALSE)
+    {
+      if(max(dplyr::pull(data, {{intensity}})) < 1000)
+      {
+      stop("Please backtransform your data or use raw values. The function does not handle log2 transformed data.")
+      }
+
+      result <- data %>%
+        dplyr::select({{grouping}}, {{condition}}, {{intensity}}) %>%
+        tidyr::drop_na({{intensity}}) %>%
+        dplyr::group_by({{grouping}}) %>%
+        dplyr::mutate(cv_combined = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::group_by({{condition}}, {{grouping}}) %>%
+        dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::select({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
+        dplyr::distinct() %>%
+        tidyr::drop_na() %>%
+        dplyr::group_by({{condition}}) %>%
+        dplyr::mutate(median_cv = stats::median(.data$cv)) %>%
+        dplyr::ungroup() %>%
+        dplyr::mutate(median_cv_combined = stats::median(.data$cv_combined)) %>%
+        dplyr::select(- {{grouping}}, - .data$cv_combined, - .data$cv) %>%
+        dplyr::distinct()
+
+      return(result)
+    }
+
+    if(plot == TRUE)
+    {
+      if(max(dplyr::pull(data, {{intensity}})) < 1000)
+      {
+        stop("Please backtransform your data or use raw values. The function does not handle log2 transformed data.")
+      }
+      result <- data %>%
+        tidyr::drop_na({{intensity}}) %>%
+        dplyr::group_by({{grouping}}) %>%
+        dplyr::mutate(cv_combined = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::group_by({{condition}}, {{grouping}}) %>%
+        dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::ungroup() %>%
+        dplyr::select({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
+        dplyr::distinct() %>%
+        tidyr::drop_na() %>%
+        tidyr::pivot_longer(cols = starts_with("cv"), names_to = "type", values_to = "values") %>%
+        dplyr::mutate(type = ifelse(.data$type == "cv", {{condition}}, "combined")) %>%
+        dplyr::select(-{{condition}}) %>%
+        dplyr::distinct()
+
+      if (max(result$values) > 200) {
+        cv_too_high <- result %>% dplyr::filter(.data$values >= 200) %>% nrow()
+        warning(paste(cv_too_high), " values were exluded from the plot (CV > 200 %)")
+      }
+
+    if(plot_style == "boxplot")
+    {
+      plot <- ggplot2::ggplot(result) +
+        ggplot2::geom_boxplot(aes(x = .data$type, y = .data$values, fill = .data$type)) +
```

Comment:
would also add a violin plot in addition to the box plot in the same plot. then you can also see the distribution of values.